platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/system/transaction/Transaction.java

```diff
@@ -53,6 +55,16 @@ public sealed interface Transaction extends SerializableWithKnownLength permits
     @NonNull
     OneOf<EventPayload.PayloadOneOfType> getPayload();
 
+    /**
+     * A convenience method for retrieving the application payload {@link Bytes} object. Before calling this method,
+     * ensure that the transaction is not a system transaction by calling {@link #isSystem()}.
```

Comment:
> Before calling this method, ensure that the transaction is not a system transaction by calling {@link #isSystem()}.

Why is this necessary, since the method does this check internally?

Reply:
Because you might expect to get a payload but will get empty Bytes